GOVERNANCE.md

```diff
@@ -0,0 +1,126 @@
+Governance for the R data.table project
+
+# Purpose and scope
+
+## This document
+
+The purpose of this document is to define how people related to the project work together, so that the project can expand to handle more contributions, from a larger and more diverse group of contributors.
+
+## The R package
+
+The purpose of the project is to maintain the R data.table package,
+which is guided by the following principles:
+
+* Few (if any) dependencies
+* Time & memory efficiency
+* Concise syntax (minimal redundancy in code)
+* Stable code base (easy for users to upgrade to new data.table, and compatible with old R versions)
+* Comprehensive and accessible documentation
+* Clear error and warning messages
+
+To prioritize developer time, we define what is in and out of scope.
+The current scope of package functionality includes:
+* Data manipulation and analysis 
+    * reshaping/pivoting
+    * aggregation/summarizing
+    * subsetting rows
+    * all sorts of joins
+    * adding/updating/deleting columns
+* Reading/writing of data from/to many file formats
+
+Functionality that is out of scope:
+* plotting/graphics (like ggplot2)
+* manipulating data stored on disk (or remote SQL DB) rather than in memory (like sqldf / dbplyr)
```

Comment:
on-disk memory mapped files are in scope

Reply:
is there any implementation of that currently? I thought there was not. Does anyone have a plan for that? If not, I would suggest to leave it out of scope, so that we can prioritize developer time (at least until some one provides an initial implementation).
There are a number of related open issues that could be closed if we decide this is out of scope
* read parquet, https://github.com/Rdatatable/data.table/issues/2026 -> this is an interesting one because it wants the `DT[i,j,by]` syntax to access a parquet file, which is a cool idea, but I would say that is out of scope for data.table itself, but could be implemented in another package.
* file-backed (csv) data tables, https://github.com/Rdatatable/data.table/issues/1336
* chunked fread https://github.com/Rdatatable/data.table/issues/1721
* etc https://github.com/Rdatatable/data.table/issues?q=is%3Aissue+is%3Aopen+disk
To be clear, I have no problem with putting "manipulating data stored on disk" in scope, if there is general consensus. But I would like to suggest we consider leaving it out of scope (at least for now, considering lack of implementation, and scarcity of dev time).

Reply:
There is no, but this is on the roadmap. Unless scope defines only current scope?

Reply:
The previous section starts:

> The current scope of package functionality includes:

So I guess this section is "currently out-of-scope". And process for changing this document is laid out below to be no-headache, so we can easily expand the scope later.

Maybe we need to make note about what sorts of "out-of-current-scope" external contributions are welcome?

Reply:
In response to scope discussions, I intended "scope" to represent what is currently implemented, and what we want to focus dev/maintenance time on. So then it is easy to close issues that are out of scope, and save dev time.

> Maybe we need to make note about what sorts of "out-of-current-scope" external contributions are welcome?

I would say that "out-of-current-scope" contributions are not welcome, so if someone wants to contribute something that is currently out of scope, they first have to get involved with the project and change scope.

Reply:
Jan do you approve like this? If not, what modifications would you suggest?

Reply:
Considering it is current scope, and not future roadmap then this point is resolved